coveralls/api.py

```diff
@@ -68,6 +64,17 @@ def ensure_token(self):
         if self.config.get('repo_token') or not self._token_required:
             return
 
+        service = self.config.get('service_name') or ''
+        if self._token_required and service.startswith('github'):
+            gh_token = os.environ.get('GITHUB_TOKEN')
+            if gh_token:
+                self.config['repo_token'] = gh_token
+                return
+            raise CoverallsException(
```

Comment:
Just to be clear with your top-level comment: what happens in the case when `GITHUB_TOKEN` is unset but `COVERALLS_REPO_TOKEN` has been provided? Should this check for either-or?

Reply:
Then load_config_from_environment sets config['repo_token'] to it and ensure_token bails out from its very first line checking for that.
Will address the rest tomorrow.

Reply:
Sorry, I should have been more clear, what I'd meant to ask was: is that Ok from Github-Actions' perspective? It sounds like both cases work (ie. either token can be used), just want to be sure.

Reply:
That depends on the service-name sent to Coveralls it seems. If it's 'github', it can be the Github-Token, if it's not 'github', i.e. the so far used name of 'github-actions', it cannot be the Github-Token.
But the behavior here is wholly undocumented, and just based on testing and guesswork.

Reply:
Gotcha, thanks for clarifying!